### DISCUSSION TITLE
Animation Disc - Rotate Canvas

### DISCUSSION THREAD

**Discussion by wwderw** [Orama-Interactive/Pixelorama#461] [Ideas]
I was thinking about something along the lines that would be a digital equivalent to an animation disc.

Krita has a feature like this in their right click context menu to where the user is able to rotate the entire canvas in order to make it easier to draw in certain areas.

Since our right click is for another tool, could either integrate as another right click tool that display some type of HUD for easy rotation and an option to clear out the rotation and go back to original position.

Or maybe a slider like what we have with Opacity, except it starts in the middle and depending when way you go on the slider rotates it clockwise or counterclockwise with being able to hit the center to quickly revert back to original orientation.

At least those were the two ideas that jump out at me for implementation.  Thoughts?

**Comment by novhack** [comment]
This feature is not even available in Aseprite so it could give Pixelorama a bit of an edge for tablet users.
@Variable-ind already made a concept in his own repository https://github.com/Variable-ind/Pixelorama/tree/canvas_rotation
UI should be discussed so it's precise with tablet pens.

**Comment by mrtripie** [comment]
This was implemented with #558 (using @Variable-ind's work as a base.
